ocf_data_sampler/torch_datasets/sample/uk_regional.py

```diff
@@ -22,22 +22,186 @@ def __init__(self, data: NumpySample) -> None:
 
     @override
     def to_numpy(self) -> NumpySample:
+        """Returns the data as a NumPy sample."""
         return self._data
 
     @override
     def save(self, path: str) -> None:
+        """Saves sample to the specified path in pickle format."""
         # Saves to pickle format
         torch.save(self._data, path)
 
     @classmethod
     @override
     def load(cls, path: str) -> "UKRegionalSample":
+        """Loads sample from the specified path.
+
+        Args:
+            path: Path to the saved sample file.
+
+        Returns:
+            A UKRegionalSample instance with the loaded data.
+        """
         # Loads from .pt format
         # TODO: We should move away from using torch.load(..., weights_only=False)
         return cls(torch.load(path, weights_only=False))
 
+    def _check_dimensions(
+        self,
+        actual_shape: tuple[int, ...],
+        expected_shape: tuple[int, ...],
+        name: str,
+    ) -> None:
+        """Check if dimensions match between actual and expected shapes.
+
+        Args:
+            actual_shape: The actual shape of the data
+            expected_shape: The expected shape
+            name: Name of the data for error messages
+
+        Raises:
+            ValueError: If dimensions don't match
+        """
+        # Check dimensions match
+        if len(actual_shape) != len(expected_shape):
+            raise ValueError(
+                f"{name} shape dimension mismatch: "
+                f"expected {len(expected_shape)} dimensions, got {len(actual_shape)}",
+            )
+
+        # Check each dimension matches
+        zipped_dims = zip(actual_shape, expected_shape, strict=False)
+        for i, (actual_dim, expected_dim) in enumerate(zipped_dims):
+            if expected_dim is not None and actual_dim != expected_dim:
+                raise ValueError(
+                    f"{name} shape mismatch at dimension {i}: "
+                    f"expected {expected_dim}, got {actual_dim}",
+                )
+
+    def validate_sample(self, config: dict | None = None) -> dict:
```

Comment:
I reckon config should be a requirement argument, and its `config:Configuration`

Reply:
see https://github.com/openclimatefix/ocf-data-sampler/blob/main/ocf_data_sampler/config/model.py#L317